### PR TITLE
[SPARK-25313][SQL][FOLLOW-UP] Fix InsertIntoHiveDirCommand output schema in Parquet issue

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

How to reproduce:
```scala
spark.sql("CREATE TABLE tbl(id long)")
spark.sql("INSERT OVERWRITE TABLE tbl VALUES 4")
spark.sql("CREATE VIEW view1 AS SELECT id FROM tbl")
spark.sql(s"INSERT OVERWRITE LOCAL DIRECTORY '/tmp/spark/parquet' " +
  "STORED AS PARQUET SELECT ID FROM view1")
spark.read.parquet("/tmp/spark/parquet").schema
scala> spark.read.parquet("/tmp/spark/parquet").schema
res10: org.apache.spark.sql.types.StructType = StructType(StructField(id,LongType,true))
```
The schema should be `StructType(StructField(ID,LongType,true))` as we `SELECT ID FROM view1`.

This pr fix this issue. 

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by wangyum** [comment]
cc @gengliangwang

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 1]
```diff
@@ -803,6 +803,23 @@ class HiveDDLSuite
     }
   }
 
+  test("Insert overwrite directory should output correct schema") {
+    withSQLConf(CONVERT_METASTORE_PARQUET.key -> "false") {
```
Comment (gengliangwang) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: Add `withTable("tbl") {` here.

**Review by gengliangwang — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by dongjoon-hyun** [comment]
Hi, @wangyum .
- Ur, I know SPARK-25313 has some information, but could you make the PR description more complete? The following PR description is just a repetition of the title. :)
```
Fix InsertIntoHiveDirCommand output schema issue.
```
- nit, `FOLLOW-UP]` -> `[FOLLOW-UP]`?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 2]
```diff
@@ -803,6 +803,25 @@ class HiveDDLSuite
     }
   }
 
+  test("Insert overwrite directory should output correct schema") {
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: Since this is a bug fix, can we have `SPARK-25313` prefix?

Comment (wangyum) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: Also added here?
https://github.com/apache/spark/blob/8e60b98239be63555644e013417cda7175baf984/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala#L758
https://github.com/apache/spark/blob/8e60b98239be63555644e013417cda7175baf984/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala#L782

Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: In this PR, let's handle this test case only.

**Comment by dongjoon-hyun** [comment]
@wangyum @gengliangwang 

Does this PR aim a correct behavior for `INSERT OVERWRITE LOCAL DIRECTORY`?
In SPARK-25313, it may make sense because the generated file are under the table and respects the table definition. However, in this PR, this accidentally introduces **case-sensitivity** in SQL SELECT statement.
```
The schema should be StructType(StructField(ID,LongType,true)) as we SELECT ID FROM view1.
```

This case seems to require a new SPARK JIRA issue and more discussion on the goal. Also, this should be allowed only when `spark.sql.caseSensitivity=true`.

cc @gatorsmile @cloud-fan

**Comment by dongjoon-hyun** [comment]
Since this is related to Parquet behavior only, can we have `in Parquet` at the end of title specifically?

**Comment by cloud-fan** [comment]
thanks, merging to master/2.4!

**Pull request closed without merging**